### ISSUE TITLE
Only import Class definitions when they are actually imported in the Ruby code

### ISSUE THREAD

**Issue by garyking** [castwide/vscode-solargraph#35]
Right now, SolarGraph imports ALL Class definitions from the current workspace. Please instead only make it work when the class is actually imported via the Ruby code, with `require`/`require_relative`.

**Comment by castwide** [comment]
Yeah, it's fair to call this long-term aspirational. It's on the roadmap, but there are a lot of cases to consider that make it decidedly non-trivial.
